src/main/java/org/threadly/concurrent/event/ListenerHelper.java

```diff
@@ -90,7 +89,7 @@ protected T makeProxyInstance(Class<? super T> listenerInterface) {
       if (listeners == null) {
         return Collections.emptyList();
       } else {
-        return Collections.unmodifiableList(new ArrayList<T>(listeners.keySet()));
+        return Collections.unmodifiableList(Pair.collectLeft(listeners));
```

Comment:
We just use it one place, I think it is a simple enough implementation that might be useful to people.